src/native/libs/System.Security.Cryptography.Native/openssl.c

```diff
@@ -964,7 +964,21 @@ int32_t CryptoNative_X509StoreSetVerifyTime(X509_STORE* ctx,
         return 0;
     }
 
+#if defined(TARGET_ARM) && defined(TARGET_LINUX)
+    // We support ARM32 linux distros that have Y2038-compatible glibc (those which support _TIME_BITS).
+    // Some such distros have not yet switched to _TIME_BITS=64 by default, so we may be running against an openssl
+    // that expects 32-bit time_t even though our time_t is 64-bit.
+
+    // Builds of openssl that use 32-bit time_t expect it to be passed to X509_VERIFY_PARAM_set_time in r1.
+    // Builds that use 64-bit time_t expect it to be passed in r2:r3, because the ARM ABI specifies:
+    // > If the argument requires double-word alignment (8-byte), the NCRN is rounded up to the next even register number.
+
+    // Let's pretend we're calling a function that takes both an int32_t and a time_t. This ensures that we set both
+    // r1 (to the low bits of our time_t) and r2:r3 (to the full time_t) which will satisfy either ABI.
+    ((void (*)(X509_VERIFY_PARAM*, int32_t, time_t))(void*)(X509_VERIFY_PARAM_set_time))(verifyParams, (int32_t)verifyTime, verifyTime);
```

Comment:
```suggestion
    // This can be deleted once the minimum supported Linux Arm32 distros are
    // at least Debian 13 and Ubuntu 24.04.
    ((void (*)(X509_VERIFY_PARAM*, int32_t, time_t))(void*)(X509_VERIFY_PARAM_set_time))(verifyParams, (int32_t)verifyTime, verifyTime);
```